From c4196a6711fd0b1fed5c9dd422ca6be97cab862f Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Mon, 18 Dec 2017 14:53:01 +0000
Subject: [PATCH 4/7] lineage-sdk: Import device hardware keys configs and
 constants

 * Used to "live" in fw/b since ever, but will be here from now on.

Change-Id: I3469a4084590c96f41c6ff128cf008143d315728
---
 lineage/res/res/values/config.xml                  | 28 +++++++++++++++++++
 lineage/res/res/values/symbols.xml                 |  4 +++
 .../internal/util/DeviceKeysConstants.java         | 31 ++++++++++++++++++++++
 3 files changed, 63 insertions(+)
 create mode 100644 sdk/src/java/org/lineageos/internal/util/DeviceKeysConstants.java

diff --git a/lineage/res/res/values/config.xml b/lineage/res/res/values/config.xml
index 8b5b9cf..cdd8d4e 100644
--- a/lineage/res/res/values/config.xml
+++ b/lineage/res/res/values/config.xml
@@ -140,4 +140,32 @@
     <string-array name="config_deviceKeyHandlerClasses" translatable="false">
         <item>org.lineageos.lineageparts.gestures.KeyHandler</item>
     </string-array>
+
+    <!-- Hardware keys present on the device, stored as a bit field.
+         This integer should equal the sum of the corresponding value for each
+         of the following keys present:
+             1 - Home
+             2 - Back
+             4 - Menu
+             8 - Assistant (search)
+            16 - App switch
+            32 - Camera
+            64 - Volume rocker
+         For example, a device with Home, Back and Menu keys would set this
+         config to 7. -->
+    <integer name="config_deviceHardwareKeys">79</integer>
+
+    <!-- Hardware keys present on the device with the ability to wake, stored as a bit field.
+         This integer should equal the sum of the corresponding value for each
+         of the following keys present:
+             1 - Home
+             2 - Back
+             4 - Menu
+             8 - Assistant (search)
+            16 - App switch
+            32 - Camera
+            64 - Volume rocker
+         For example, a device with Home, Back and Menu keys would set this
+         config to 7. -->
+    <integer name="config_deviceHardwareWakeKeys">79</integer>
 </resources>
diff --git a/lineage/res/res/values/symbols.xml b/lineage/res/res/values/symbols.xml
index 026b4c4..3aea364 100644
--- a/lineage/res/res/values/symbols.xml
+++ b/lineage/res/res/values/symbols.xml
@@ -116,6 +116,10 @@
     <java-symbol type="array" name="config_deviceKeyHandlerLibs" />
     <java-symbol type="array" name="config_deviceKeyHandlerClasses" />
 
+    <!-- Device hardware keys -->
+    <java-symbol type="integer" name="config_deviceHardwareKeys" />
+    <java-symbol type="integer" name="config_deviceHardwareWakeKeys" />
+
     <!-- BurnIn protection -->
     <java-symbol type="bool" name="config_enableBurnInProtection" />
 
diff --git a/sdk/src/java/org/lineageos/internal/util/DeviceKeysConstants.java b/sdk/src/java/org/lineageos/internal/util/DeviceKeysConstants.java
new file mode 100644
index 0000000..053afa9
--- /dev/null
+++ b/sdk/src/java/org/lineageos/internal/util/DeviceKeysConstants.java
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2017 The LineageOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.lineageos.internal.util;
+
+public class DeviceKeysConstants {
+    // Masks for checking presence of hardware keys.
+    // Must match values in:
+    //   lineage/res/res/values/config.xml
+    public static final int KEY_MASK_HOME = 0x01;
+    public static final int KEY_MASK_BACK = 0x02;
+    public static final int KEY_MASK_MENU = 0x04;
+    public static final int KEY_MASK_ASSIST = 0x08;
+    public static final int KEY_MASK_APP_SWITCH = 0x10;
+    public static final int KEY_MASK_CAMERA = 0x20;
+    public static final int KEY_MASK_VOLUME = 0x40;
+}
+
-- 
2.7.4

